core/src/com/unciv/logic/map/mapunit/MapUnit.kt

```diff
@@ -997,7 +998,11 @@ class MapUnit : IsPartOfGameInfoSerialization {
                 civ.getDifficulty().clearBarbarianCampReward * civ.gameInfo.speed.goldCostModifier
         if (civ.hasUnique(UniqueType.TripleGoldFromEncampmentsAndCities))
             goldGained *= 3f
-
+        
+        for (unique in civ.getMatchingUniques(UniqueType.GoldFromEncampmentsAndCities)) {
```

Comment:
``` suggestion
        for (unique in civ.getMatchingUniques(UniqueType.GoldFromEncampmentsAndCities, cache.state)) {
```
While this does change the original behavior, passing in the unit's state will mean you could do things like...

```
Receive [+500]% Gold from Barbarian encampments and pillaging Cities <for [Barbarian Plunderer] units>
```